pkg/endpoint/metrics.go

```diff
@@ -65,10 +66,10 @@ func (s *regenerationStatistics) SendMetrics() {
 		// Skip scopes that have not been hit (zero duration), so the count in
 		// the histogram accurately reflects the number of times each scope is
 		// hit, and the distribution is not incorrectly skewed towards zero.
-		if stat.SuccessTotal() != time.Duration(0) {
+		if stat.SuccessTotal() > time.Duration(0) {
 			metrics.EndpointRegenerationTimeStats.WithLabelValues(scope, "success").Observe(stat.SuccessTotal().Seconds())
 		}
-		if stat.FailureTotal() != time.Duration(0) {
+		if stat.FailureTotal() > time.Duration(0) {
```

Comment:
Do we still need these changes?

Logically if somehow a spanstat doesn't get `End()`ed due to another bug, then presumably this would still silently discard metrics?